### PR TITLE
Feat/training-datasets

### PR DESCRIPTION
**Pull request by Eysteinn-Orn**
added 1 folder, 2 files: "__init__.py" and "config.py" and a small class to __init__.py.
Config.py my be unnecessary at the current time.

### CODE REVIEW

File: training_datasets/__init__.py [hunk 1]
```diff
@@ -1,4 +1,18 @@
-from torch.utils.data import DataLoader, Dataset
-
 import torch
-import random
\ No newline at end of file
+
+
+class SquadDataset(torch.utils.data.Dataset):
```
Comment (njallskarp) [inline training_datasets/__init__.py:4]: Refactor this

**Review by njallskarp — CHANGES_REQUESTED** [review]
Create a datasets.py module and put the Dataset class in there

**Comment by eysta00** [comment]
organized imports*

**Comment by lsig** [comment]
I was wondering if you could format the docstrings so they are inline, otherwise looks good. Maybe I can setup a formatter.

File: training_datasets/datasets.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    """
+    class SquadDataset(Dataset):
+        """
+        A PyTorch dataset.
+
+        Args:
+            encodings (any): The preprocessed input encodings.
+            is_train (bool, optional): Whether the dataset is for training or not. Defaults to True.
+        """
+    def __init__(self, encodings: any, is_train: bool = True):
```
Comment (njallskarp) [inline training_datasets/datasets.py:18]: Add types, check in with @lsig if you need details on the types.

**Review by njallskarp — CHANGES_REQUESTED** [review]
see comment. Need to improve typing

File: training_datasets/datasets.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import torch
+from torch.utils.data import Dataset
+
+
+class SquadDataset(Dataset):
+    """
+    Code reused from open source colab notebook:
```
Comment (njallskarp) [inline training_datasets/datasets.py:7]: Class should not be nested, please review and fix

**Review by njallskarp — CHANGES_REQUESTED** [review]
fix nesting of classes

**Comment by eysta00** [comment]
@lsig what is the type for encodings?

**Comment by eysta00** [comment]
> Collaborator
@lsig https://github.com/njallskarp/finetune-qa-powerset/pull/9#issuecomment-1609587840
are you talking about docstring comments beginning like this:
`""" Some text begins here

Args:
   The arguments

Returns:
    arguments added together`

File: training_datasets/datasets.py (hunk 2 above)
Comment (lsig) [reply training_datasets/datasets.py:10]: Let me check

Comment (lsig) [reply training_datasets/datasets.py:10]: It is the return type of tokenizer

Comment (lsig) [reply training_datasets/datasets.py]: Probably Tokenzier().encodings or something like that

**Pull request closed without merging**